WordPress/Classes/System/3DTouch/WP3DTouchShortcutHandler.swift

```diff
@@ -0,0 +1,70 @@
+import UIKit
+
+public class WP3DTouchShortcutHandler: NSObject
+{
+    enum ShortcutIdentifier: String {
+        case LogIn
+        case NewPost
+        case NewPhotoPost
+        case Stats
+        case Notifications
+        
+        init?(fullType: String) {
+            guard let last = fullType.componentsSeparatedByString(".").last else { return nil }
```

Comment:
Great call. There actually seems to actually already be a pseudo-standard, e.g. (in Gravatar.swift):

```
guard let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) else {
            return false
        }

guard let sanitizedURL = components.URL else {
            return nil
        }
```

Changing!